common/src/main/java/com/wynntils/screens/maps/PoiManagementScreen.java

```diff
@@ -67,79 +111,756 @@ public void onClose() {
+            this.removeWidget(widget);
+        }
+
+        this.poiManagerWidgets.clear();
+
+        // Get full list of pois
+        pois = Managers.Feature.getFeatureInstance(MainMapFeature.class).customPois.get().stream()
+                .filter(poi -> searchMatches(poi.getName()))
+                .collect(Collectors.toList());
+
+        // No pois
+        if (pois.isEmpty()) {
+            return;
+        }
+
+        // Filter pois based on filtered icons
+        pois = pois.stream()
+                .filter(poi -> filteredIcons.getOrDefault(poi.getIcon(), true))
+                .collect(Collectors.toList());
+
+        // Hide buttons if no filtered pois
+        iconSortButton.visible = !pois.isEmpty();
+        nameSortButton.visible = !pois.isEmpty();
+        xSortButton.visible = !pois.isEmpty();
+        ySortButton.visible = !pois.isEmpty();
+        zSortButton.visible = !pois.isEmpty();
+        exportButton.active = !pois.isEmpty();
+        selectAllButton.active = !pois.isEmpty();
+        deselectAllButton.active = !selectedPois.isEmpty();
+
+        // Only hide search bar & filter button if no pois at all
+        // Can't check if filtered pois is empty as they may be empty due to the filters
+        if (Managers.Feature.getFeatureInstance(MainMapFeature.class)
+                .customPois
+                .get()
+                .isEmpty()) {
+            searchInput.visible = false;
+            filterButton.visible = false;
+        }
+
+        // No filtered pois
+        if (pois.isEmpty()) {
+            return;
+        }
+
+        // Sort pois
+        if (sortOrder != null) {
+            pois = sortPois();
+        }
+
+        // Starting Y position for the pois
+        int row = (int) ((int) (dividedHeight * HEADER_HEIGHT) + (dividedHeight / 2f));
+
+        int currentPoi;
+
+        // Render manager widgets
+        for (int i = 0; i < maxPoisToDisplay; i++) {
+            // Get the poi to render
+            currentPoi = i + scrollOffset;
+
+            // If user has less pois than maxPoisToDisplay, make sure we don't try and get a poi out of range
+            if (currentPoi > pois.size() - 1) {
+                break;
+            }
+
+            CustomPoi poi = pois.get(currentPoi);
+
+            PoiManagerWidget poiWidget = new PoiManagerWidget(
+                    (int) (dividedWidth * 12),
+                    row,
+                    (int) (dividedWidth * 38),
+                    20,
+                    poi,
+                    this,
+                    dividedWidth,
+                    selectionMode,
+                    selectedPois.contains(poi));
+
+            // Each widget height is 20, add 20 for Y position of next widget
+            row += 20;
+
+            poiManagerWidgets.add(poiWidget);
+
+            this.addRenderableWidget(poiWidget);
+        }
+    }
+
+    private List<CustomPoi> sortPois() {
+        List<CustomPoi> sortedPois = pois;
```

Comment:
You don't actually copy the list so you just sort `pois`. If that is intended, I am not sure why this method returns anything.